biz.aQute.junit.platform.reporting/bnd.bnd

```diff
@@ -0,0 +1,8 @@
+# Set javac settings from JDT prefs
+-include: ${workspace}/cnf/includes/jdt.bnd
+
+-includeresource.reporting: @${repo;org.junit.platform:junit-platform-reporting;${junit-platform-reporting}}
+-includeresource.launcher: @${repo;org.junit.platform:junit-platform-launcher;${junit-platform-reporting}}!/org/junit/platform/launcher/listeners/LegacyReportingUtils.class
+-exportcontents: org.junit.platform.reporting.*;version="${junit-platform-reporting}"
+Bundle-Version: ${junit-platform-reporting}
```

Comment:
All the Bnd bundles must be the same version (e.g. 4.3.0 for this release) as set in cnf. I don't see any reason why this bundle should be different.

Reply:
This was supposed to simply be a repackaging of `org.junit.platform:junit-platform-reporting` to add an OSGi manifest and turn it into a bundle. Hence I gave it the same version as the jar that it was repackaging (just like `org.apache.servicemix.bundles.junit` has version 4.12).

All of the other JUnit 5 jars that I needed already came ready-bundled as part of Eclipse. This one was not. This bundles provides an out-of-the-box implementation of the JUnit Platform's `TestExecutionListener` that generates legacy JUnit XML files. I saw no need to re-invent the wheel.

Because the main Eclipse bundle of `junit-platform-launcher` was on 1.0.x, and 1.4.0 of this junit-platform-reporting required a class from junit-platform-launcher:1.4.0, I also copy that class into this bundle to decouple this bundle from that version and make it more self-contained. This way this bundle will work even with 1.0.0 of junit-platform-launcher.

Perhaps this bundle really probably belongs properly somewhere else - eg, Orbit (with the rest of the JUnit 5 bundles that have been bundled for Eclipse), or enRoute (like `osgi.enroute.junit.wrapper`).

Reply:
It probably should come from somewhere else. But all the Bnd bundles in the build are the same version. We build the each build and release them each release. So I don't really want something different here. Perhaps it makes sense to just conditional package the necessary junit-platform-reporting packages into our bundle as an impl detail?

Reply:
A problem with conditional packaging it in is that it uses `org.junit.platform.launcher.listeners.LegacyReportingUtils`. This class only became available in 1.4.0 of junit-platform-launcher. But `biz.aQute.tester.junit-platform` also uses other classes from the same package, so I can't conditional package Legacy ReportingUtils into the bundle without dragging in those classes as well, or else the bundle will need to bind to v>=1.4.0 of junit-platform-launcher, which means we can't use the version that comes prepackaged with Eclipse Oxygen (which is on 1.0.0). This will cascade to all the other junit-platform bundles.

Maybe it would be better to go the enRoute route?

Reply:
> Maybe it would be better to go the enRoute route?

Except enRoute does not do that (wrap bundles) anymore. enRoute v7 is maven based build env. The old v2 enroute is no longer maintained or supported by OSGi any more.

https://github.com/junit-team/junit5/blob/master/junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/LegacyReportingUtils.java does not seem a complex class. Surely we could write our own replacement to avoid this dependency altogether.

Reply:
Or maybe I am not understanding what we need from junit-platform-reporting...

Reply:
You're right, LegacyReportingUtils is not a complex class. The problem is not that we can't replace it (we don't even use it directly), the problem is that LegacyXmlReportGeneratingListener uses it. If we rewrite LegacyReportingUtils, we'll have to rewrite LegacyXmlReportGeneratingListener too so that it links to our version of the class. I actually did this when I was debugging - cut-and-pasted the source code into `biz.aQute.junit.platform` and moved LegacyReportingUtils into another package to avoid the collision/spilt package. For maintenance reasons I'd like to avoid that, do that our bundle can be used with different versions of Junit 5